project/frontend/src/App.js

```diff
@@ -36,57 +41,57 @@ function App() {
     setCompId(parsedCompId);
   }, []);
 
-  if (user.signedIn && !user.id) {
-    //TODO: fetch for id... with useremail param... once backend is ready
-    //for now id hardcoded to 123
-    setUser((prevState) => {
+  const fetchCompetitionInfo = async (fetchCall, stateKey) => {
+    try {
+      const data = await fetch(fetchCall).then((response) => response.json());
+
       return {
-        ...prevState,
-        id: 1,
+        [stateKey]: data,
       };
-    });
-  }
-
-  const fetchAndUpdateCompetitionInfo = async (fetchCall, stateKey) => {
-    return fetch(fetchCall)
-      .then((response) => response.json())
-      .then((data) => {
-        setCompetitionInfo((prevState) => {
-          return {
-            ...prevState,
-            [stateKey]: data,
-          };
-        });
-      });
+    } catch (e) {
+      return {
+        [stateKey]: undefined,
+      };
+    }
   };
 
   React.useEffect(() => {
+    if (!authStateReceived) {
+      return;
+    }
+
     if (user.signedIn && user.id && compId) {
-      Promise.all([
-        setLoading(true),
-        fetchAndUpdateCompetitionInfo(
-          "./competitionInfo?user=" + user.id + "&competition=" + compId,
-          "generalInfo"
-        ),
-        fetchAndUpdateCompetitionInfo(
-          "./recentBuys?competition=" + compId,
-          "recentBuys"
-        ),
-        fetchAndUpdateCompetitionInfo(
-          "./investments?user=" + user.id + "&competition=" + compId,
-          "investments"
-        ),
-        fetchAndUpdateCompetitionInfo("./trending", "trending"),
-      ])
-        .then(() => {
+      setLoading(true),
+        Promise.all([
+          fetchCompetitionInfo(
+            "./competitionInfo?user=" + user.id + "&competition=" + compId,
+            "generalInfo"
+          ),
+          fetchCompetitionInfo(
+            "./recentBuys?competition=" + compId,
+            "recentBuys"
+          ),
+          fetchCompetitionInfo(
+            "./investments?user=" + user.id + "&competition=" + compId,
+            "investments"
+          ),
+          fetchCompetitionInfo("./trending", "trending"),
+        ]).then((resolvedData) => {
           setLoading(false);
+          console.log(resolvedData);
           console.log(competitionInfo);
-        })
-        .catch(() => {
-          setLoading(false);
+          let newCompInfo = {};
+          for (const response of resolvedData) {
+            newCompInfo = {
+              ...newCompInfo,
+              ...response,
+            };
+          }
+
+          setCompetitionInfo(newCompInfo);
         });
     }
-  }, [user.id]);
+  }, [user.id, authStateReceived]);
 
   let pageRoute = !user.signedIn ? (
```

Comment:
I'd like to move away from using ternaries that are over 1 or 2 short lines as I think it doesn't convey intent as clearly.
I think initialising to a default value and then testing conditions by priority is more readable - brief example
```js
let pageContent = <div>Signing in...</div>

if (authStateReceived && !user.SignedIn) {
  pageContent = (
    <Layout>
      // ...
```

This logic is also long enough that it could be extracted into another function, perhaps this should be a child component?

Reply:
Refactored :D